### ISSUE TITLE
Missing error for `Final` in top level of type alias statement

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#3150]
### Summary

[Playground](https://play.ty.dev/3b1c4b43-f893-4397-ac49-10aade0c1024)
```py
from typing import Final
# Expected error: Type qualifiers are not allowed in type alias definitions (invalid-type-form)
type A = Final[int]  # No error
```
This should give an error, like how the manual type alias version does https://play.ty.dev/8f204899-4720-44bb-b9bd-c86d72f86aed

### Version

playground 229e49f20
